pkg/kncloudevents/message_sender_test.go

```diff
@@ -194,6 +196,345 @@ func TestHTTPMessageSenderSendWithRetriesWithSingleRequestTimeout(t *testing.T)
 	require.Equal(t, http.StatusOK, got.StatusCode)
 }
 
+func TestHTTPMessageSenderSendWithRetriesWithRetryAfter(t *testing.T) {
+	t.Parallel()
+
+	// Create A Quick Enum For RetryAfter Format
+	type RetryAfterFormat int
+	const (
+		None RetryAfterFormat = iota
+		Seconds
+		Date
+		Invalid
+	)
+
+	// Test Data
+	smallRetryAfterMaxDuration := 1 * time.Second
+	largeRetryAfterMaxDuration := 10 * time.Second
+
+	// Define The TestCases
+	tests := []struct {
+		name            string
+		config          *RetryConfig     // The RetryConfig to use in for the test case.
+		statusCode      int              // The HTTP StatusCode which the Server should return.
+		responseFormat  RetryAfterFormat // Indicates the format in which the Server should return the Retry-After header.
+		responseSeconds int              // Indicates the number of Seconds for the Server to use when returning the Retry-After header.
+		wantReqCount    int              // The total number of expected requests, including initial request.
+	}{
+		//
+		// Default Max Tests (0 Duration opt-in vs opt-out)
+		//
+
+		{
+			name: "default max 429 without Retry-After",
+			config: &RetryConfig{
+				RetryMax: 2,
+			},
+			statusCode:     http.StatusTooManyRequests,
+			responseFormat: None,
+			wantReqCount:   3,
+		},
```

Comment:
I'm not sure I understand the comment. :smile:

Reply:
Agreed - I'll clarify the comment - just means the "default" case where no maxRetryAfter is specified.